File: 030)MyNewDatabase/app/src/main/java/com/elyesasimsek/mynewdatabase/MainActivity.java
package com.elyesasimsek.mynewdatabase;

import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;
import android.os.Bundle;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;
import androidx.databinding.DataBindingUtil;

import com.elyesasimsek.mynewdatabase.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {

    private SQLiteDatabase db;
    private ActivityMainBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        binding = DataBindingUtil.setContentView(this, R.layout.activity_main);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });

        try {
            db = this.openOrCreateDatabase("Family", MODE_PRIVATE, null);
            //db.execSQL("CREATE TABLE IF NOT EXISTS family (name VARCHAR, age INT(2))");
            //db.execSQL("INSERT INTO family (name, age) VALUES ('Elyesa', 25)");
            db.execSQL("DROP TABLE IF EXISTS family");

            db.execSQL("CREATE TABLE family (uniqueId INTEGER PRIMARY KEY AUTOINCREMENT, name VARHAR, age Int(2))");

            db.execSQL("INSERT INTO family (name, age) VALUES ('Elyesa', 25)");
            db.execSQL("INSERT INTO family (name, age) VALUES ('Fatma', 25)");

            Cursor cursor = db.rawQuery("SELECT * FROM family", null);

            int idIx = cursor.getColumnIndex("uniqueId");
            int nameIx = cursor.getColumnIndex("name");
            int ageIx = cursor.getColumnIndex("age");

            while (cursor.moveToNext()){
                System.out.println("Name: " + cursor.getString(nameIx));
                System.out.println("Age: " + cursor.getString(ageIx));
                System.out.println("Id: " + cursor.getString(idIx));
            }
            cursor.close();
        }catch (Exception e){
            e.printStackTrace();
        }
    }
}

File: 052)RunnableKotlin/app/src/main/java/com/elyesasimsek/runnablekotlin/MainActivity.kt
package com.elyesasimsek.runnablekotlin

import android.os.Bundle
import android.os.Handler
import android.os.Looper
import android.view.View
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import androidx.databinding.DataBindingUtil
import com.elyesasimsek.runnablekotlin.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding
    private var number = 0
    var runnable : Runnable = kotlinx.coroutines.Runnable {  }
    var handler: Handler = Handler(Looper.getMainLooper())

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        binding = DataBindingUtil.setContentView(this, R.layout.activity_main)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }
    }

    fun start(view: View){
        runnable = object : Runnable{
            override fun run() {
                number++
                binding.textView.text = "Time: ${number}"
                handler.postDelayed(runnable, 1000)
            }
        }
        handler.post(runnable)
        binding.button.isEnabled = false
    }

    fun stop(view: View){
        binding.button.isEnabled= true
        number = 0
        binding.textView.text = "Time: ${number}"
        handler.removeCallbacks(runnable)
    }
}